dotCMS/src/main/java/com/dotmarketing/portlets/hostvariable/bussiness/HostVariableAPIImpl.java

```diff
@@ -157,6 +160,23 @@ public List<HostVariable> save(final List<HostVariable> siteVariables, final Str
 		return processedVariables;
 	}
 
+	/**
+	 * Retrieves a list of unique site variables from the given list of site variables.
+	 *
+	 * @param siteVariables The list of site variables.
+	 * @return A new ArrayList containing only the unique site variables based on their keys.
+	 */
+	private ArrayList<HostVariable> getUniqueSiteVariables(List<HostVariable> siteVariables) {

```

Comment:
Perhaps a bit too picky but you should return the interface List not the implementation ArrayList

Reply:
or even better return an immutable List rather than an ArrayList

Reply:
Good feedback Fabro, thanks, I applied your suggestions.